plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiPageSourceProvider.java

```diff
@@ -0,0 +1,141 @@
+import io.trino.plugin.hudi.page.HudiPageSourceBuilder;
+import io.trino.plugin.hudi.page.HudiPageSourceBuilderFactory;
+import io.trino.plugin.hudi.page.HudiParquetPageSourceBuilder;
+import io.trino.spi.block.Block;
+import io.trino.spi.connector.ColumnHandle;
+import io.trino.spi.connector.ConnectorPageSource;
+import io.trino.spi.connector.ConnectorPageSourceProvider;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.ConnectorSplit;
+import io.trino.spi.connector.ConnectorTableHandle;
+import io.trino.spi.connector.ConnectorTransactionHandle;
+import io.trino.spi.connector.DynamicFilter;
+import io.trino.spi.predicate.Utils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hudi.common.model.HoodieFileFormat;
+import org.joda.time.DateTimeZone;
+
+import javax.inject.Inject;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+import java.util.stream.Collectors;
+
+import static io.trino.plugin.hudi.HudiUtil.convertPartitionValue;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
+
+public class HudiPageSourceProvider
+        implements ConnectorPageSourceProvider
+{
+    private final HudiConfig hudiConfig;
+    private final HdfsEnvironment hdfsEnvironment;
+    private final FileFormatDataSourceStats fileFormatDataSourceStats;
+    private final DateTimeZone timeZone;
+    private final Map<HoodieFileFormat, HudiPageSourceBuilder> pageSourceBuilderMap;
+    private final Map<String, Object> context;
+
+    @Inject
+    public HudiPageSourceProvider(
+            HdfsEnvironment hdfsEnvironment,
+            FileFormatDataSourceStats fileFormatDataSourceStats,
+            ParquetReaderConfig parquetReaderConfig,
+            HudiConfig hudiConfig)
+    {
+        this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
+        this.fileFormatDataSourceStats = requireNonNull(fileFormatDataSourceStats, "fileFormatDataSourceStats is null");
+        this.hudiConfig = requireNonNull(hudiConfig, "hudiConfig is null");
+        this.timeZone = DateTimeZone.forID(TimeZone.getDefault().getID());
+        this.pageSourceBuilderMap = new HashMap<>();
+        this.context = new HashMap<>();
+        this.context.put(
+                HudiParquetPageSourceBuilder.CONTEXT_KEY_PARQUET_READER_OPTIONS,
+                requireNonNull(parquetReaderConfig, "parquetReaderConfig is null").toParquetReaderOptions());
+    }
+
+    @Override
+    public ConnectorPageSource createPageSource(
+            ConnectorTransactionHandle transaction,
+            ConnectorSession session,
+            ConnectorSplit connectorSplit,
+            ConnectorTableHandle connectorTable,
+            List<ColumnHandle> columns,
+            DynamicFilter dynamicFilter)
```

Comment:
Can you point me to where the file pruning happens i.e filtering out files that don't match the filter ?